packages/next/src/lib/metadata/resolve-metadata.ts

```diff
@@ -294,24 +272,37 @@ export async function accumulateMetadata(
 
       // This resolve should unblock the generateMetadata function if it awaited the parent
       // argument. If it didn't await the parent argument it might already have a value since it was
-      // called concurrently. Regardless we await the return value before continuing on to the next
-      // layer
+      // called concurrently. Regardless we await the return value before continuing on to the next layer
       resolveParent(currentResolvedMetadata)
-      metadata = await generatedMetadata
+      metadata =
+        generatedMetadata instanceof Promise
+          ? await generatedMetadata
+          : generatedMetadata
     } else {
       metadata = metadataExport
     }
 
     // If the layout is the same layer with page, skip the
-    if (isLeafLayout && metadata) {
-      delete metadata.title
+    if ((i === 0 || (i !== 0 && i !== metadataItems.length - 2)) && metadata) {
```

Comment:
Does this distinguish between these cases

case 1
```
/app/layout.js
/app/page.js
```

case 2
```
/app/layout.js
/app/some/page.js
```

case 3
```
/app/layout.js
/app/some/really/deep/page.js
```

case 4
```
/app/layout.js
/app/some/layout.js
/app/some/deep/path/page.js // imagine this page does not export any metadata at all
```

case 1: page.js should not have a template scoped because there is no "higher up" layout.
case 2 and case 3: have the same behavior, the title in page.js is templatized with the template in layout even though the layout is potentially many folder levels higher up.
case 4: the page doesn't export metadata so the title in some/layout should be templatized by the template in /layout

Reply:
Yeah those are covered, always the closest and non adjacent parent layout title templated will be picked up. 1,2,3 were covered. Also added a test case for case 4 in this PR already

Reply:
I misunderstood `metadataItems`. I see now that it always has a slot for every layer + the page. the strategy here makes sense